src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java

```diff
@@ -16,394 +16,559 @@
-		stmtGetStock =this.getPreparedStatement(conn, stmtGetStockSQL);
-		stmtUpdateStock =this.getPreparedStatement(conn, stmtUpdateStockSQL);
-		stmtInsertOrderLine =this.getPreparedStatement(conn, stmtInsertOrderLineSQL);
-
-
-		int districtID = TPCCUtil.randomNumber(terminalDistrictLowerID,terminalDistrictUpperID, gen);
-		int customerID = TPCCUtil.getCustomerID(gen);
-
-		int numItems = (int) TPCCUtil.randomNumber(5, 15, gen);
-		int[] itemIDs = new int[numItems];
-		int[] supplierWarehouseIDs = new int[numItems];
-		int[] orderQuantities = new int[numItems];
-		int allLocal = 1;
-		for (int i = 0; i < numItems; i++) {
-			itemIDs[i] = TPCCUtil.getItemID(gen);
-			if (TPCCUtil.randomNumber(1, 100, gen) > 1) {
-				supplierWarehouseIDs[i] = terminalWarehouseID;
-			} else {
-				do {
-					supplierWarehouseIDs[i] = TPCCUtil.randomNumber(1,
-							numWarehouses, gen);
-				} while (supplierWarehouseIDs[i] == terminalWarehouseID
-						&& numWarehouses > 1);
-				allLocal = 0;
-			}
-			orderQuantities[i] = TPCCUtil.randomNumber(1, 10, gen);
-		}
-
-		// we need to cause 1% of the new orders to be rolled back.
-		if (TPCCUtil.randomNumber(1, 100, gen) == 1)
-			itemIDs[numItems - 1] = TPCCConfig.INVALID_ITEM_ID;
-
-
-		newOrderTransaction(terminalWarehouseID, districtID,
-						customerID, numItems, allLocal, itemIDs,
-						supplierWarehouseIDs, orderQuantities, conn, w);
-		return null;
+  public final SQLStmt stmtUpdateStockSQL = new SQLStmt(
+      "UPDATE " + TPCCConstants.TABLENAME_STOCK +
+      " SET S_QUANTITY = ?, S_YTD = ?, S_ORDER_CNT = S_ORDER_CNT + 1,  S_REMOTE_CNT = ? " +
+      "WHERE S_W_ID = ? AND S_I_ID = ?");
+
+  public SQLStmt[] stmtGetItemSQL;
+  public SQLStmt[] stmtGetStockSQL;
+  public SQLStmt[] stmtInsertOrderLineSQL;
+
+
+  // NewOrder Txn
+  private PreparedStatement stmtGetCust = null;
+  private PreparedStatement stmtGetWhse = null;
+  private PreparedStatement stmtGetDist = null;
+  private PreparedStatement stmtInsertNewOrder = null;
+  private PreparedStatement stmtUpdateDist = null;
+  private PreparedStatement stmtInsertOOrder = null;
+  private PreparedStatement stmtUpdateStock = null;
+  private PreparedStatement stmtInsertOrderLine = null;
+  private PreparedStatement stmtGetItem = null;
+  private PreparedStatement stmtGetStock = null;
+
+  public NewOrder() {
+    stmtGetItemSQL = new SQLStmt[15];
+    stmtGetStockSQL = new SQLStmt[15];
+    stmtInsertOrderLineSQL = new SQLStmt[11];
+
+    // We create 15 statements for selecting rows from the `ITEM` table with varying number of ITEM
+    // ids.  Each string looks like:
+    // SELECT I_ID, I_PRICE, I_NAME , I_DATA
+    // FROM ITEM
+    // WHERE I_ID in (?, ? ..);
+    StringBuilder sb = new StringBuilder();
+    sb.append(String.format("SELECT I_ID, I_PRICE, I_NAME , I_DATA FROM %s WHERE I_ID in (",
+                            TPCCConstants.TABLENAME_ITEM));
+    for (int ii = 1; ii <= 15; ++ii) {
```

Comment:
nit: `ii` -> `i`

Reply:
Done.
Back in nutanix single letter variables were not allowed and we used 'ii' and 'jj' as the index variables. Hence the habit.